docs/tutorial/t-kafka-connect.md

```diff
@@ -0,0 +1,383 @@
+juju integrate postgresql-connect-integrator postgresql
+juju integrate postgresql-connect-integrator kafka-connect
+```
+
+After a couple of minutes, `juju status` command should show the `postgresql-connect-integrator` in `active|idle` state, with a message indicating that the ETL task is running:
+
+```text
+...
+postgresql-connect-integrator/0*  active    idle   13       10.38.169.83    8080/tcp  Task Status: RUNNING
+...
+```
+
+This means that the integrator application is actively copying data from the source database, i.e. `tutorial` into Apache Kafka topics prefixed with `etl_`. For example, rows in the `posts` table will be published into an Apache Kafka topic named `etl_posts`.
+
+## 6. Deploy and integrate the `opensearch-connect-integrator` charm
+
+You are almost done with the ETL task, the only remaining part is to move data from Apache Kafka to Opensearch. To achieve that, you need to deploy another Kafka Connect integrator named `opensearch-connect-integrator` in `sink` mode:
+
+```bash
+juju deploy opensearch-connect-integrator \
+    --channel edge \
+    --config mode=sink \
+    --config topics="etl_posts"
+```
+
+The above command would deploy an integrator application to move messages published into the `etl_posts` topic to an index in Opensearch named `etl_posts`. We know that the `etl_posts` topic is being filled by the `postgresql-connect-integrator` charm we deployed in step 5.
+
+To activate the `opensearch-connect-integrator`, make the necessary integrations:
+
+```bash
+juju integrate opensearch-connect-integrator opensearch
+juju integrate opensearch-connect-integrator kafka-connect
+```
+
+Wait a couple of minutes and run `juju status`, now both `opensearch-connect-integrator` and `postgresql-connect-integrator` applications should be in `active|idle` state, showing a message indicating that the ETL task is running:
+
+```text
+...
+opensearch-connect-integrator/0*  active    idle   14       10.38.169.108   8080/tcp  Task Status: RUNNING
+postgresql-connect-integrator/0*  active    idle   13       10.38.169.83    8080/tcp  Task Status: RUNNING
+...
+```
+
+## 7. Verify the data is being copied
+
+Now it's time to verify that the data is being copied from the PostgreSQL database to the Opensearch index. We can use the Opensearch REST API for that purpose.
+
+First, retrieve the admin user credentials for Opensearch using `get-password` action:
+
+```bash
+juju run opensearch/leader get-password
+```
+
+Sample output:
+```text
+...
+password: GoCNE5KdFywT4nF1GSrwpAGyqRLecSXC
+username: admin
+```
+
+Now, using the password obtained above, send a request to the topic's `_search` endpoint, either using your browser or `curl`:
+
+```bash
+curl -u admin:<admin-password> -k -X GET https://<opensearch-unit-ip>:9200/etl_posts/_search
+```
+
+You will get a JSON response containing the search results, which should have 5 documents. A truncated sample output is shown below (Note the `hits.total` value which should be 5):
```

Comment:
Suggestion: Wording improvements for clarity. Also, we should try to avoid using digits like 1-9 in the usual text. Ideally, replace them with words or literal values (in backticks).

```suggestion
As a result you get a JSON response containing the search results, which should have five documents. 
The `hits.total` value should be `5`, as shown in the output example below:
```